File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.py
class Solution:
    def maxDifference(self, s: str) -> int:
        counter = Counter(s)
        mx = -maxsize
        mn = maxsize

        for freq in counter.values():
            if freq & 1:
                mx = max(mx, freq)
            else:
                mn = min(mn, freq)

        return mx - mn

File: 3753-maximum-difference-between-even-and-odd-frequency-i/maximum-difference-between-even-and-odd-frequency-i.cpp
class Solution {
public:
    int maxDifference(string s) {
        vector<int> freq(26, 0);

        for (char c : s) {
            freq[c - 'a']++;
        }

        int mx = INT_MIN;
        int mn = INT_MAX;

        for (int count : freq) {
            if (count > 0) {
                if (count % 2 == 0) {
                    mn = min(mn, count);
                } else {
                    mx = max(mx, count);
                }
            }
        }

        return mx - mn;
    }
};